azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/NetworkSecurityRule.java

```diff
@@ -177,182 +182,196 @@ public static Protocol fromString(String s) {
     }
 
     /**
-     * The entirety of a network security rule definition as part of a network security group update.
-     * @param <ParentT> the return type of the final {@link DefinitionAttachable#attach()}
-     */
-    interface UpdateDefinition<ParentT> extends
-        UpdateDefinables.Blank<ParentT>,
-        UpdateDefinables.WithDirectionAccess<ParentT>,
-        UpdateDefinables.WithSourceAddress<ParentT>,
-        UpdateDefinables.WithSourcePort<ParentT>,
-        UpdateDefinables.WithDestinationAddress<ParentT>,
-        UpdateDefinables.WithDestinationPort<ParentT>,
-        UpdateDefinables.WithProtocol<ParentT>,
-        UpdateDefinables.WithAttach<ParentT> {
-    }
-
-    /**
-     * The entirety of a security rule update as part of a network security group update.
+     * Grouping of security rule definition stages applicable as part of a network security group creation.
      */
-    interface Update extends
-        Updatables.WithDirectionAccess,
-        Updatables.WithSourceAddress,
-        Updatables.WithSourcePort,
-        Updatables.WithDestinationAddress,
-        Updatables.WithDestinationPort,
-        Updatables.WithProtocol,
-        Settable<NetworkSecurityGroup.Update> {
-
+    interface Definables {
         /**
-         * Specifies the priority to assign to this rule.
-         * <p>
-         * Security rules are applied in the order of their assigned priority.
-         * @param priority the priority number in the range 100 to 4096
-         * @return the next stage of the update
+         * The first stage of a security rule definition.
+         * @param <ParentT> the return type of the final {@link DefinitionAttachable#attach()}
          */
-        Update withPriority(int priority);
-    }
+        interface Blank<ParentT> extends WithDirectionAccess<ParentT> {
+        }
 
-    /**
-     * Grouping of security rule update stages.
-     */
-    interface Updatables {
         /**
-         * The stage of the network rule description allowing the direction and the access type to be specified.
+         * The stage of the security rule definition allowing the protocol that the rule applies to to be specified.
+         * @param <ParentT> the return type of the final {@link DefinitionAttachable#attach()}
```

Comment:
We can enable this in eclipse http://help.eclipse.org/mars/index.jsp?topic=%2Forg.eclipse.jdt.doc.user%2Freference%2Fpreferences%2Fjava%2Fcompiler%2Fref-preferences-javadoc.htm


Preferences -> Java -> Compiler -> Javadoc

Check "Process Javadoc comments"
Set "Malformed Javadoc comments" as "Warning"
Check "Validate tag arguments"